### PR TITLE
Modifications to the command line tool

### PR DESCRIPTION
**Pull request by surilindur**
Here is a PR with the following changes, that can be further split up if needed:
* Remove positional parameters from the CLI to make it easier to see exactly what parameters are provided, with the exception of the query that can still be read from standard input.
* Add timeout to the CLI tool as an option, since it exists in the library already.
* Add support for basic username and password authentication to the CLI, which should only be used for testing purposes because the password ends up in the command prompt...

The following changes were made to the CI:
* Change to use service containers for the Jena test with Docker for the CLI, which removes the need to manually start or stop the SPARQL endpoint, or provide a name for it.
* Run the CLI test with a SPARQL endpoint for each Node version by including it in the test phase after unit tests. This should help ensure that all tested Node versions also work in practice with the CLI.

This work is a follow-up for what has been done in #74 and may need to be rebased after that is merged. Any feedback would be welcome, though.

### CODE REVIEW

**Comment by rubensworks** [comment]
Could you ping me once this PR has been rebased?

**Comment by surilindur** [comment]
This has been rebased now, and should be easier to review @rubensworks. Any feedback is welcome, and I can fix things that seem out of place or unnecessary.

**Comment by surilindur** [comment]
I also added a command line tool test for ASK queries that should hopefully pass the CI.

File: README.md [hunk 1]
```diff
@@ -141,22 +141,31 @@ This method will also throw an error if the query contains a syntax error.
 
 ### Command-line
 
-A command-line tool is provided to quickly query or update any SPARQL endpoint:
+A command-line tool is provided to quickly query or update any SPARQL endpoint.
+Basic authentication is meant for development or debugging only.
```
[the remark this answers is not in the document]
Comment (rubensworks) [reply README.md:145]: That may be better indeed. Because if we offer this functionality, people _will_ use it :-)

**Comment by surilindur** [comment]
> Are the docker changes still needed now? Because that means we can't easily debug using docker locally now.

Would simply restoring the bash scripts be enough for that? As in, not deleting them in the first place.

The use of service containers is a lot cleaner solution for the CI, though, since it handles the pulling of images, starting and stopping on its own. It also makes it easier to run the tests with Docker for each Node version, without having to keep track of which container ID is running for which version.

**Comment by rubensworks** [comment]
> Would simply restoring the bash scripts be enough for that? As in, not deleting them in the first place.

Yes, it would. But if we want to be sure that they keep working in the future, we'd have to run them in the CI as well. But then the question rises whether it make sense to run the docker tests in two different ways.

**Comment by surilindur** [comment]
There is now a bash script that allows starting and stopping the Jena container, and running the tests. It seems to work finally in the CI, as well. :sweat:

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (surilindur) [reply README.md]: I have changed the code to use environment variables when `--auth basic` is set in the command line, so this should be resolved.

**Comment by rubensworks** [comment]
Super, thanks @surilindur!

**Pull request merged**